darts/models/forecasting/torch_forecasting_model.py

```diff
@@ -1244,6 +1244,138 @@ def lr_find(
+            override Darts' default trainer.
+        verbose
+            Optionally, whether to print the progress. Ignored if there is a `ProgressBar` callback in
+            `pl_trainer_kwargs`.
+        epochs
+            If specified, will train the model for ``epochs`` (additional) epochs, irrespective of what ``n_epochs``
+            was provided to the model constructor.
+        max_samples_per_ts
+            Optionally, a maximum number of samples to use per time series. Models are trained in a supervised fashion
+            by constructing slices of (input, output) examples. On long time series, this can result in unnecessarily
+            large number of training samples. This parameter upper-bounds the number of training samples per time
+            series (taking only the most recent samples in each series). Leaving to None does not apply any
+            upper bound.
+        num_loader_workers
+            Optionally, an integer specifying the ``num_workers`` to use in PyTorch ``DataLoader`` instances,
+            both for the training and validation loaders (if any).
+            A larger number of workers can sometimes increase performance, but can also incur extra overheads
+            and increase memory usage, as more batches are loaded in parallel.
+        method
+            The method to use for scaling the batch size. Can be one of 'fit', 'validate', 'test', or 'predict'.
+        mode
+            The mode to use for scaling the batch size. Can be one of 'power' or 'linear'.
+        steps_per_trial
+            The number of steps to try for each trial.
+        init_val
+            The initial value to start the search with.
+        max_trials
+            The maximum number of trials to run.
+        batch_arg_name
+            The name of the argument to scale in the model. Defaults to 'batch_size'.
+
+        Returns
+        -------
+        int
+            The largest batch size that can be used without running out of memory.
+        """
+        _, params = self._setup_for_fit_from_dataset(
+            series=series,
+            past_covariates=past_covariates,
+            future_covariates=future_covariates,
+            val_series=val_series,
+            val_past_covariates=val_past_covariates,
+            val_future_covariates=val_future_covariates,
+            trainer=trainer,
+            verbose=verbose,
+            epochs=epochs,
+            max_samples_per_ts=max_samples_per_ts,
+            num_loader_workers=num_loader_workers,
+        )
+        trainer, model, train_loader, val_loader = self._setup_for_train(*params)
+
+        class DataModule(pl.LightningDataModule):
+            def __init__(self, batch_size):
+                super().__init__()
+                self.save_hyperparameters()
+                self.batch_size = batch_size
+
+            def train_dataloader(self):
+                return DataLoader(
```

Comment:
since we use this also in `_setup_for_train` and `predict_from_dataset`, it would be good to have this logic in a private method for example `_build_dataloader()`, that takes as input a dataset, and returns the dataloader according to "train", "val" and "predict"